components/prism-sas.js

```diff
@@ -10,25 +23,152 @@ Prism.languages.sas = {
 			'punctuation': /;/
 		}
 	},
+	'procSQL': {
+		pattern: /(^|\n)(?:proc\s+sql(?:\s+[\w|=]+)?);([\s\S]+(?:\r?\n|\r)(?=((?:proc\s\w+|quit|run|data(?!\=));))((?:proc\s\w+|quit|run|data(?!\=));)|(?:\r?\n|\r)[\s\S]+;)/im,
+		inside: {
+			'fullStep': {
+				pattern: /(^|\n)(?:proc\s+sql(?:\s+[\w|=]+)?);/i,
+				inside: {
+					'step': {
+						pattern: /(?:proc\s\w+|quit|run|data(?!\=))\b/i,
+						alias: 'keyword'
+					},
+					'argument': {
+						pattern: /\w+(?==)/,
+						alias: 'keyword'
+					},
+					'operator': /=/,
+					'punctuation': /;/,
+					'number': numberpattern,
+					'numeric-constant': numericconstant,
+					'string': string
+				}
+			},
+			'step': {
+				pattern: /(?:proc\s\w+|quit|run|data(?!\=))\b/i,
+				alias: 'keyword'
+			},
+			rest: Prism.languages.sql
+		}
+	},
+	'macrodeclaration': {
+		pattern: /((^|\n)(%macro)((\s)+([\s\S]+?))+);/im,
+		inside: {
+			keyword: /%macro/im,
+		}
+	},
+	'macroend': {
+		pattern: /^%mend[\s\S]+?;/i,
```

Comment:
You accidentally delete the `m` flag.
```suggestion
		pattern: /^%mend[\s\S]+?;/im,
```

Reply:
I actually intended to remove that. the pattern is generally 

> %mend macroName;

I'm not sure that there's ever a line break.

Reply:
`^` with the `m` flag means start of line. But without the `m` flag it means start of string.
The `m` flag is essential.

Here is a macro highlighted with the current pattern without the `m` flag.
![image](https://user-images.githubusercontent.com/20878432/61716704-fc6cd500-ad5f-11e9-89a0-f20673055947.png)
[Source](http://support.sas.com/documentation/cdl/en/mcrolref/61885/HTML/default/viewer.htm#macro-stmt.htm).

As you can see, it's highlighted incorrectly.

Please make a test for macros in general (you can use the above code if you want) and remove the ones for `macro-declaration` and `macro-end`.

Also as a bit of general advice: Try to get your hands on as much SAS code as possible (>1k lines) and see how it gets highlighted. This will make problems like the missing punctuation visible.

Reply:
Yes, thanks. I am testing against SAS code, and directly comparing the results of Prism.JS highlighting vs. SAS Studio.